### DISCUSSION TITLE
Radiomaster TX12 and 6 Flight Modes using trim buttons

### DISCUSSION THREAD

**Discussion by DangerD1024** [EdgeTX/edgetx#3660] [Q&A]
I have Radiomaster TX12 and i want to setup 6 Flight modes using trims:
![image](https://github.com/EdgeTX/edgetx/assets/4560434/c2705c5d-f47f-4fee-b351-460b4f277734)
Here i can set just one switch (not combination):
![image](https://github.com/EdgeTX/edgetx/assets/4560434/c7d07864-8e57-40cc-8460-9eb3c444df09)
So looks like i have to move everything to Logical Switches...
Can someone point me on that?

**Comment by pfeerick** [comment]
> Here i can set just one switch (not combination):

What exactly do you mean by this? Do you want to use more than one switch to set the FM? e.g. SC and/or SB?

You'll have to use logical switches anyway, as you can't use momentary switches for your flight modes (well, you can, if you want to keep the button held down all the time you want the FM active). i.e. `tTu` is 'trim Throttle up', so in your example, FM4 will be active as long as you are holding the throttle trim upwards. 

You could overcome this with a sticky switch, that looks at the state of tTu, and first time you toggle it, it latches, and the second time you toggle it, it releases. But you then probably also need logic to see if any other logical switch in that set is active, and clear it if you try to change flight modes. It is probably doable, but is probably going to get quite messy. It may be better to use a GV instead to set the active flight mode... i.e. when you do a trim throttle up, it uses a SF to set the GV to 1, which then makes a LS true,  which the FM monitors for as it's switch, And when you then do a throttle trim down, it would set the GV to 2, which makes the LS for the acro FM active, and thus the FM would change again... 

i.e. these special functions would change the value of global variable when those trims were pressed... I added the seventh entry so that flight mode 0 could be "manual" mode. 

![image](https://github.com/EdgeTX/edgetx/assets/5500713/60b917fb-f941-4156-96d9-1f2f8494d9ad)

Then, on the logical switches page, you would monitor the GV value, so that the logical switch is made active. This is needed as you need a switch to change your flight modes ;) Technically `L07` is not actually needed, but it's there for completeness. i.e. you might want to also use the LS states as the trigger for a special function to play a corresponding track to say which flight mode is active... 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/3fa6fb98-c4be-4566-ae3a-354523dedd16)

So finally back at the flight modes page, you can see that FM1 - FM6 are triggered by a logical switch. FM0 is special as it's the "default" flight mode, so as long as none of the other flight modes are active, it will be. This is why LS 7 was technically not needed.
![image](https://github.com/EdgeTX/edgetx/assets/5500713/0cadb3e6-638f-4ada-9376-26fb8be426c6)

Now, there is one thing I have not done here, which is to inhibit the trims in the various flight modes. You really want to disable the trims for all but manual mode, so that when you activate the flight modes, you don't also alter the trims for those channels. This won't alter the ability to use them to trigger the flight modes, just their behaviour as channel trims.
